### PR TITLE
Avoid deepcopying defaults for containers in stage1

### PR DESCRIPTION
**Pull request by maxnoe**

### CODE REVIEW

**Review by kosack — DISMISSED** [review]

**Review by kosack — APPROVED** [review]

**Review by kosack — DISMISSED** [review]

**Review by kosack — APPROVED** [review]

**Review by LukasNickel — DISMISSED** [review]

**Comment by maxnoe** [comment]
@kosack @LukasNickel had to merge, please re-approve

**Review by kosack — DISMISSED** [review]

File: ctapipe/containers.py [hunk 1]
```diff
@@ -403,9 +403,7 @@ class SimulatedCameraContainer(Container):
         ndim=1,
     )
 
-    true_parameters = Field(
-        ImageParametersContainer(), "Parameters derived from the true_image"
-    )
+    true_parameters = Field(None, "Parameters derived from the true_image")
```
Comment (HealthyPear) [inline ctapipe/containers.py:406]: `type=ImageParametersContainer)` also here?

File: ctapipe/image/image_processor.py [hunk 2]
```diff
@@ -143,8 +148,8 @@ def _parameterize_image(
                     container_class=PeakTimeStatisticsContainer,
                 )
             else:
-                timing = TimingParametersContainer()
-                peak_time_statistics = PeakTimeStatisticsContainer()
+                timing = DEFAULT_TIMING_PARAMETERS
+                peak_time_statistics = DEFAULT_PEAKTIME_STATISTICS
```
Comment (HealthyPear) [inline ctapipe/image/image_processor.py:152]: Codecov is complaining that these 2 lines are not covered by unit-testing

File: ctapipe/io/simteleventsource.py [hunk 3]
```diff
@@ -430,23 +432,25 @@ def _generate_events(self):
...
         azimuth_cor = tracking_position.get("azimuth_cor", np.nan)
         altitude_cor = tracking_position.get("altitude_cor", np.nan)
 
         # take pointing corrected position if available
         if np.isnan(azimuth_cor):
-            pointing.azimuth = u.Quantity(azimuth_raw, u.rad)
+            azimuth = u.Quantity(azimuth_raw, u.rad, copy=False)
         else:
-            pointing.azimuth = u.Quantity(azimuth_cor, u.rad)
+            azimuth = u.Quantity(azimuth_cor, u.rad, copy=False)
```
Comment (HealthyPear) [inline ctapipe/io/simteleventsource.py:445]: also this line seems to not be called by any test

File: ctapipe/io/simteleventsource.py [hunk 4]
```diff
@@ -430,23 +432,25 @@ def _generate_events(self):
...
 
         # take pointing corrected position if available
         if np.isnan(altitude_cor):
-            pointing.altitude = u.Quantity(altitude_raw, u.rad)
+            altitude = u.Quantity(altitude_raw, u.rad, copy=False)
         else:
-            pointing.altitude = u.Quantity(altitude_cor, u.rad)
+            altitude = u.Quantity(altitude_cor, u.rad, copy=False)
```
Comment (HealthyPear) [inline ctapipe/io/simteleventsource.py:451]: and this one (basically the same thing to check)

Comment (maxnoe) [reply ctapipe/io/simteleventsource.py:451]: Can you provide a simtel file that has pointing corrections?

File: ctapipe/image/image_processor.py (hunk 2 above)
Comment (maxnoe) [reply ctapipe/image/image_processor.py:152]: That could be done by testing the stage1 tool on a file with true images. But let's leave that to another PR

**Comment by maxnoe** [comment]
@kosack @HealthyPear I adressed the comments, please re-review

File: ctapipe/io/simteleventsource.py [hunk 5]
```diff
@@ -430,23 +432,25 @@ def _generate_events(self):
...
+    def _fill_event_pointing(tracking_position):
         azimuth_raw = tracking_position["azimuth_raw"]
         altitude_raw = tracking_position["altitude_raw"]
         azimuth_cor = tracking_position.get("azimuth_cor", np.nan)
         altitude_cor = tracking_position.get("altitude_cor", np.nan)
 
         # take pointing corrected position if available
         if np.isnan(azimuth_cor):
-            pointing.azimuth = u.Quantity(azimuth_raw, u.rad)
+            azimuth = u.Quantity(azimuth_raw, u.rad, copy=False)
```
Comment (kosack) [inline ctapipe/io/simteleventsource.py:443]: These are just floats, so I guess copy=False has no effect (doesn't hurt either I suppose).

**Review by kosack — APPROVED** [review]

File: ctapipe/io/simteleventsource.py (hunk 5 above)
Comment (maxnoe) [reply ctapipe/io/simteleventsource.py:443]: I think there are 0d numpy arrays, let me check.

**Review by HealthyPear — APPROVED** [review]

File: ctapipe/io/simteleventsource.py [hunk 6]
```diff
@@ -430,23 +432,25 @@ def _generate_events(self):
             yield data
 
     @staticmethod
-    def _fill_event_pointing(pointing, tracking_position):
+    def _fill_event_pointing(tracking_position):
```
Comment (vuillaut) [inline ctapipe/io/simteleventsource.py:435]: this is not really a "fill" method anymore, is it?

**Review by vuillaut — APPROVED** [review]
Just a minor comment, but this is really nit-picking, feel free to address it or not.

File: ctapipe/io/simteleventsource.py (hunk 6 above)
Comment (maxnoe) [reply ctapipe/io/simteleventsource.py:435]: depends. Fills the container...

**Pull request merged**